### PR TITLE
SDK-2306: Migrate findbugs (3.0.5) → spotbugs (4.7.3)

### PR DESCRIPTION
**Pull request by irotech**

### CODE REVIEW

File: yoti-sdk-parent/pom.xml [hunk 1]
```diff
@@ -225,16 +226,18 @@
...
-          <artifactId>findbugs-maven-plugin</artifactId>
-          <version>${findbugs-maven-plugin.version}</version>
+          <groupId>com.github.spotbugs</groupId>
+          <artifactId>spotbugs-maven-plugin</artifactId>
+          <version>${spotbugs-maven-plugin.version}</version>
           <configuration>
+            <addSourceDirs>true</addSourceDirs>
+            <spotbugsXmlOutputDirectory>${basedir}/target/spotbugs</spotbugsXmlOutputDirectory>
+            <spotbugsXmlOutputFilename>report.xml</spotbugsXmlOutputFilename>
+            <excludeFilterFile>${basedir}/spotbugs/exclude-filter.xml</excludeFilterFile>
```
Comment (pn-santos) [inline yoti-sdk-parent/pom.xml:238]: `project.basedir` is preferable to `basedir`

File: yoti-sdk-spring-boot-auto-config/spotbugs/exclude-filter.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<FindBugsFilter>
+
+  <Match>
+    <Bug pattern="CRLF_INJECTION_LOGS"/>
```
Comment (pn-santos) [inline yoti-sdk-spring-boot-auto-config/spotbugs/exclude-filter.xml:6]: when excluding something, it's a good practice to explain why with a comment - is it a bug? a false positive? a rule that can be safely ignored in the particular project? e.g.
```xml
<!-- These patterns became completely broken after 4.3 - ignored them -->
<!-- See https://github.com/spotbugs/spotbugs/issues/1797 -->
<Bug pattern="EI_EXPOSE_REP,EI_EXPOSE_REP2"/>
```

Comment (irotech) [reply yoti-sdk-spring-boot-auto-config/spotbugs/exclude-filter.xml:6]: Sure. I didn't add any new exceptions tho, just renaming the legacy XMLs. Now that the build doesn't fail with the spotbugs violations I was almost about to drop all the exceptions to start from scratch. I'll see if I can add few.

Comment (pn-santos) [reply yoti-sdk-spring-boot-auto-config/spotbugs/exclude-filter.xml:6]: ah ok the diff looked like it was new nvm mind me then :+1:

**Review by pn-santos — APPROVED** [review]

**Review by pn-santos — APPROVED** [review]

**Pull request merged**